ikarus/finiteelements/feresulttypes.hh

```diff
@@ -0,0 +1,146 @@
+                                                                                                   \
+    using type       = Eigen::Matrix<ScalarType, rowsExpr, ColsExpr>;                              \
+    using Vectorizer = VectorizeStruct;                                                            \
+    using Matricizer = MatricizeStruct;                                                            \
+  }
+
+#define REGISTER_SYMMETRIC_RESULTTYPE(structName, rowsExpr, ColsExpr)                        \
+  REGISTER_RESULTTYPE_IMPL(structName, rowsExpr, ColsExpr, Ikarus::Util::VectorizeWithVoigt, \
+                           Ikarus::Util::MatricizeWithVoigt)
+
+#define REGISTER_RESULTTYPE(structName, rowsExpr, ColsExpr)                                \
+  REGISTER_RESULTTYPE_IMPL(structName, rowsExpr, ColsExpr, Ikarus::Util::VectorizeGeneric, \
+                           Ikarus::Util::MatricizeGeneric)
+
+  ///=============================
+  /// Registered ResultTypes
+
+  REGISTER_SYMMETRIC_RESULTTYPE(linearStress, worldDim, worldDim);
+  REGISTER_SYMMETRIC_RESULTTYPE(PK2Stress, worldDim, worldDim);
+  REGISTER_SYMMETRIC_RESULTTYPE(cauchyStress, worldDim, worldDim);
+
+  REGISTER_RESULTTYPE(director, worldDim, 1);
+  REGISTER_RESULTTYPE(magnetization, worldDim, 1);
+  REGISTER_RESULTTYPE(gradientNormOfMagnetization, 1, 1);
+  REGISTER_RESULTTYPE(vectorPotential, worldDim, 1);
+  REGISTER_RESULTTYPE(divergenceOfVectorPotential, 1, 1);
+
+  REGISTER_RESULTTYPE(BField, 1, 1);
+  REGISTER_RESULTTYPE(HField, 1, 1);
+
+  REGISTER_RESULTTYPE(customType, Eigen::Dynamic, Eigen::Dynamic);
+} // namespace ResultType
+
+template <template <typename, int, int> class RT1, template <typename, int, int> class RT2>
+constexpr bool isSameResultType = std::is_same_v<RT1<double, 1, 1>, RT2<double, 1, 1>>;
+
+template <typename RT, bool inputIsVec>
+struct ResultTypeContainer : RT
+{
+private:
+  using RegisteredType = typename RT::type;
+  using Traits         = typename RegisteredType::CompileTimeTraits;
+  using ScalarType     = typename RegisteredType::Scalar;
+  static constexpr bool registeredTypeIsDynamic =
+      Traits::RowsAtCompileTime == Eigen::Dynamic or Traits::ColsAtCompileTime == Eigen::Dynamic;
+  static constexpr bool registeredTypeIsVector = Traits::IsVectorAtCompileTime;
+
+public:
+  using VecType    = std::conditional_t<registeredTypeIsDynamic, Eigen::MatrixX<ScalarType>,
+                                     std::invoke_result_t<typename RT::Vectorizer, RegisteredType>>;
+  using MatType    = std::conditional_t<registeredTypeIsDynamic, Eigen::MatrixX<ScalarType>,
+                                     std::invoke_result_t<typename RT::Matricizer, VecType>>;
+  using InputType  = std::conditional_t<inputIsVec, VecType, RegisteredType>;
+  using ResultType = RT;
+
+  const InputType& operator()() const { return value_; }
+  const InputType& operator()() { return value_; }
```

Comment:
The pair of these operators does not make sense.

The second one has to be a mutable reference
```cpp
const InputType& operator()() const { return value_; }
  InputType& operator()() { return value_; }

Reply:
But I wonder if this is really needed. The first one may be sufficent